fdbclient/SystemData.cpp

```diff
@@ -748,8 +759,7 @@ const KeyRef tagThrottleCountKey = LiteralStringRef("\xff\x02/throttledTags/manu
 // Client status info prefix
 const KeyRangeRef fdbClientInfoPrefixRange(LiteralStringRef("\xff\x02/fdbClientInfo/"),
                                            LiteralStringRef("\xff\x02/fdbClientInfo0"));
-const KeyRef fdbClientInfoTxnSampleRate = LiteralStringRef("\xff\x02/fdbClientInfo/client_txn_sample_rate/");
-const KeyRef fdbClientInfoTxnSizeLimit = LiteralStringRef("\xff\x02/fdbClientInfo/client_txn_size_limit/");
+// See remaining fields in GlobalConfig.actor.h
```

Comment:
How are we handling backward compatibility with these keys? There are two questions -- one is what happens to these keys on upgrade, and the other is whether users of old API versions should be allowed to use these keys to access the configuration of these options. Possibly it's not a big deal for old API versions, but we should make sure that users of this functionality have a reasonable story.

Reply:
Hmm, this is a good point, I'm not sure what to do here. On upgrade these keys will immediately change to ones pointing into the global configuration keyspace, and existing values in the database for these old keys will not be read anymore. This could break existing workloads that have set these keys to some non-default value and expect them to remain the same after an upgrade.

A client using these constants to read the value of these keys would also break, since they are now defined without the special key space prefix. One way to handle this issue would be to leave these constants as is, create new constants specifically for the values when stored in global configuration, and add a special conditional when setting these keys that also sets them using these old constants whenever they are set using global configuration. It's not a super elegant solution but it would solve the problems you mentioned. Otherwise, any upgrade is going to break previous functionality using these keys. What do you think of that solution?

Reply:
I think the thing that's most important to have happen is that we should preserve the values of this configuration when upgrading, if possible. If we were willing to give up maintaining these keys, then we could do that by reading in the existing values, applying them to the global config, and then deleting them. Another alternative would be to insert some extra marker that indicates they've been upgraded.

I think we have at least 3 options for what to do with the old keys:

1. Just get rid of them
2. Keep them synchronized with the global config (one-way or two-way)
3. Delete the keys, but re-route point reads or writes to the global config. This could also be guarded by API versioning. Range reads would be more awkward to implement.

I don't feel super strongly that we need to make perfect compatibility with keys that live in the system key-space. Maybe we should loop in @xumengpanda and team to comment on this, since they might be impacted by this. The high level summary is that we would be moving the location of the configuration for the client sampling profiler. If there is any tooling that accesses the system keys for this directly, then they could be affected. Accessing this data through the `fdbcli` functionality should still work.

Reply:
I added special keys for these two system keys.
Looks like this framework will replace them, I guess those code need to be changed or deprecated and use this framework.
@sfc-gh-ljoswiak You can ignore this for now, I am just commenting here to follow up the final design decision here, then I can make the change related to this.

Reply:
@brownleej @apkar Do we have any prod tooling based on these two keys?

Reply:
Yes, we do.

Reply:
To what extent are you going to require server compatibility with existing tooling during an upgrade opposed to handling the behavior change in the tooling? What kind of accesses do you have (e.g. point reads, point writes, any range reads)? This helps to evaluate between the possible choices we have compatibility wise. I believe `fdbcli` has commands to work with these keys and those would likely be easy to make compatible between versions; is it feasible to use that in your tooling?

Reply:
We do reads, writes, and clears on specific keys. I'll take a look at the feasibility of switching our tools to use the specific CLI commands, since that sounds like the better way to go in general.

Reply:
Added #4643 to track the removal of these keys. We can add backwards compatibility if necessary before the 7.0 release.